giraffe/README.md

```diff
@@ -362,17 +362,25 @@ When using the comma separated values (CSV) from the Flux query as the `fluxResp
   - "linear" scaling means the same distance between ticks represents the same increase in value.
   - "log" (logarithmic) scaling means the same distance between ticks can represent an exponential increase in value, used as a way to display data with a very wide range of values in a compact space.
 
-- **xDomain**: _array[min, max]. Optional._ The x domain of the plot can be explicitly set with numbers denoting a minimum and a maximum value for the x-axis. If this option is passed, both _min_ and _max_ are required to be numbers, making the `<Plot>` operate in a "controlled" mode, where it always uses the passed x domain to set the minimum and maximum value of the x-axis. Any brush interaction with the `<Plot>` that should change the x domain will call the `onSetXDomain` option when the component is in controlled mode. Double clicking the plot will call `onResetXDomain`. If the `xDomain` option is not passed, then the component is "uncontrolled". It will compute, set, and reset the `xDomain` automatically.
+- **xDomain**: _array[min, max]. Optional. When used, must also include at least 2 additional properties. 1) Must also include **onSetXDomainDefault** or **onSetXDomain**, or both. 2) Must also include **onResetXDomainDefault** or **onResetXDomain**, or both._ The x domain of the plot can be explicitly set with numbers denoting a minimum and a maximum value for the x-axis. If this option is passed, both _min_ and _max_ are required to be numbers, making the `<Plot>` operate in a "controlled" mode, where it always uses the passed x domain. Any brush interaction with the `<Plot>` that should change the x domain will use Giraffe's built in default handlers if `onSetXDomainDefault` is `true`. Additionally, the `onSetXDomain` callback function will be invoked if provided. Double clicking the plot will call Giraffe's built in default handers if `onResetXDomainDefault` is `true`. Additionally, the `onResetXDomain` callback function will be invoked if provided. If the `xDomain` option is not passed, then the component is "uncontrolled". It will compute, set, and reset the `xDomain` automatically.
```

Comment:
Is it possible to configure the default handlers by only passing in a function, rather than a function and a boolean? Is it possible to infer the intent from the value of the function either being a function or `undefined`?

I'm thinking of this from the perspective of having to modify the state correctly to have this work

Reply:
Yes, it is possible just to pass in a function. Or just the boolean, or even both. They are independent of each other. The behavior here is additive.

Reply:
We need both because there may be a use case where it is desirable to suppress the default handlers and just use the callback function. The callback function is not a real handler though, because it doesn't have access to private fields in `PlotEnv`